system_metrics_collector/src/system_metrics_collector/linux_process_cpu_measurement_node.hpp

```diff
@@ -0,0 +1,94 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef SYSTEM_METRICS_COLLECTOR__LINUX_PROCESS_CPU_MEASUREMENT_NODE_HPP_
+#define SYSTEM_METRICS_COLLECTOR__LINUX_PROCESS_CPU_MEASUREMENT_NODE_HPP_
+
+#include <chrono>
+#include <cmath>
+#include <string>
+#include <tuple>
+
+#include "../../src/system_metrics_collector/periodic_measurement_node.hpp"
+#include "../../src/system_metrics_collector/proc_cpu_data.hpp"
+#include "../../src/system_metrics_collector/utilities.hpp"
+
+#include "rclcpp/rclcpp.hpp"
+#include "rcutils/logging_macros.h"
+
+
+namespace system_metrics_collector
+{
+
+/**
+ * Measures the CPU percentage used by the process.
+ */
+class LinuxProcessCpuMeasurementNode : public PeriodicMeasurementNode
+{
+public:
+  /**
+   * Constructs a LinuxProcessCpuMeasurementNode.
+   *
+   * @param name the name of this node
+   * @param measurement_period the period of this node, used to read measurements
+   * @param topic the topic name used for publishing
+   * @param publish_period the period at which metrics are published.
+   */
+  LinuxProcessCpuMeasurementNode(
+    const std::string & name,
+    const std::chrono::milliseconds measurement_period,
+    const std::string & topic,
+    const std::chrono::milliseconds publish_period);
+
+protected:
+  /**
+   * Creates ROS2 timers and a publisher for periodically triggering measurements and publishing
+   * MetricsMessages.
+   * This function needs to be overridden because this class needs to also reset last_measurement_
```

Comment:
Just a note: this comment was very helpful when I wondered why ```last_measurement_``` wasn't initialized in the constructor.